src/tools/lib_fs_utils.rs

```diff
@@ -173,12 +173,17 @@ fn cp_into_lind(cage: &Cage, hostfilepath: &interface::RustPath, lindfilepath: &
 
     let lindfd = cage.open_syscall(lindtruepath.to_str().unwrap(), O_CREAT | O_TRUNC | O_WRONLY, S_IRWXA);
     assert!(lindfd >= 0);
-    let veclen = filecontents.len();
-    let fileslice = filecontents.as_slice();
-    let writtenlen = cage.write_syscall(lindfd, fileslice.as_ptr(), veclen);
 
+    let veclen = filecontents.len();
+    let mut writtenlen: usize = 0;
+    //on Linux, write() (and similar system calls) will transfer at most 0x7ffff000 (2,147,479,552) bytes
+    const LINUX_MAX_RW_COUNT: usize = 2147479552;
```

Comment:
I'd move this to the top of the file, maybe declare it using the exponent function for easier reading with a comment